src/librustc/ty/context.rs

```diff
@@ -718,19 +716,17 @@ impl<'tcx> TypeckTables<'tcx> {
         }
     }
 
-    pub fn cast_kinds(&self) -> LocalTableInContext<'_, ty::cast::CastKind> {
-        LocalTableInContext {
-            local_id_root: self.local_id_root,
-            data: &self.cast_kinds
+    pub fn is_coercion_cast(&self, hir_id: &hir::HirId) -> bool {
+        if self.coercion_casts.contains(&hir_id.local_id) {
```

Comment:
simplify the body of this function to just `self.coercion_casts.contains(&hir_id.local_id)`